### PR TITLE
test lift

### PR DESCRIPTION
**Pull request by franvila**
### Type of change

- Enhancement / new feature

### Description

In order to have a build with no errors in Kroxylicious we would need to avoid java 19 jdk (`integrationtests` module) because there is no current support in Sonatype for that version. The LTS supported is JDK 17.

### Additional Context

_Why are you making this pull request?_

### CODE REVIEW

**Comment by k-wall** [comment]
@franvila I clicked ready for review by mistake and couldn't find a way to switch it back.

**Comment by k-wall** [comment]
> The only thing that I don't know how to skip is to avoid running ./mvnw by default when it is present in the project,

Looks like you could config what mvnw does by having your setup script create `$HOME/.mavenrc ` (https://github.com/franvila/kproxy/blob/lift/mvnw#L43). I think you could use it to set `MAVEN_CONFIG` to `-pl !:integrationtests --also-make-dependents`.

File: scripts/setupLift.sh
Comment (k-wall) [inline scripts/setupLift.sh:15]: interactive true seems like an odd choice

```suggestion
  <interactiveMode>false</interactiveMode>
```

File: scripts/setupLift.sh
Comment (k-wall) [inline scripts/setupLift.sh:14]: Avoids assuming `~` is `/home/lift` (untested).


```suggestion
M2_HOME=~/.m2
mkdir -p ${M2_HOME}
cat <<EOF > ${M2_HOME}/settings.xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <localRepository>${M2_HOME}/repository</localRepository>
```

File: .lift.toml
Comment (k-wall) [inline .lift.toml:2]: AFAIK Lift is static analysis, so running the tests just adds latency without benefit.

```suggestion
build = "mvn install -pl !:integrationtests --also-make-dependents -DskipTests"
```

Comment (franvila) [reply .lift.toml:2]: Yes, I don't mean to merge this PR, this was just to play around with lift application and help Sam's PR.

**Comment by franvila** [comment]
> @franvila I clicked ready for review by mistake and couldn't find a way to switch it back.

Once it is reviewed it cannot be switched back. This PR was intentionally created in draft to not merge it. The original one is that one from Sam, #242

File: scripts/setupLift.sh
Comment (franvila) [reply scripts/setupLift.sh]: Tested and working fine. Thanks @k-wall !!

File: scripts/setupLift.sh
Comment (franvila) [reply scripts/setupLift.sh]: done

**Review by k-wall — APPROVED** [review]
looks reasonable to me.

**Comment by franvila** [comment]
> > The only thing that I don't know how to skip is to avoid running ./mvnw by default when it is present in the project,
> 
> Looks like you could config what mvnw does by having your setup script create `$HOME/.mavenrc ` (https://github.com/franvila/kproxy/blob/lift/mvnw#L43). I think you could use it to set `MAVEN_CONFIG` to `-pl !:integrationtests --also-make-dependents`.

@k-wall tested but it does not work. It seems setup and ./mvnw are executed in parallel so they are not always executed in the same order and does not affect the other.

**Comment by k-wall** [comment]
How about ignoring mvnw files in our tree with a `ignoreFiles` config setting?  Like this:

https://community.sonatype.com/t/dependency-tab-using-maven-should-be-gradle/10323/3?u=kwall

**Comment by franvila** [comment]
@k-wall @SamBarker My intention was not to merge this PR, but if you agree we can merge this PR and close Sam's #242

**Comment by franvila** [comment]
> How about ignoring mvnw files in our tree with a `ignoreFiles` config setting? Like this:
> 
> https://community.sonatype.com/t/dependency-tab-using-maven-should-be-gradle/10323/3?u=kwall

I tried that as well, but nothing. It seems it is configured by default to detect mvnw files and .lift.toml and execute the process in parallel if they files are found, without ignoring any of those if the other is found.

**Comment by k-wall** [comment]
> > How about ignoring mvnw files in our tree with a `ignoreFiles` config setting? Like this:
> > https://community.sonatype.com/t/dependency-tab-using-maven-should-be-gradle/10323/3?u=kwall
> 
> I tried that as well, but nothing. It seems it is configured by default to detect `mvnw` files and `.lift.toml` and execute the process in parallel if they files are found, without ignoring any of those if the other is found.
> 
> Anyway, it is "only" noise, non time consuming as they are executed in parallel, we just need one compilation to be able to run the static analysis.

You are right, it probably doesn't matter as the build is reasonable quick. I suppose setup script could run.

```
rm -f mvnw*
```

**Comment by franvila** [comment]
> rm -f mvnw*

Tested in this [commit](https://github.com/kroxylicious/kroxylicious/pull/244/commits/c141a0e55b5d818d92a3be70ac0ea8b34d9c4ebd) with no luck https://lift.sonatype.com/results/github.com/kroxylicious/kroxylicious/01GWVQP2KNRBCHQN9YMXRVYE55/?ph=PhaseIntroduced&tab=logs.

As I suspected, both processes are run in parallel without knowing each other, so no matter what the setup does, if mvnw exists at the beginning it will be launched by default.

File: scripts/setupLift.sh
Comment (SamBarker) [reply scripts/setupLift.sh]: That confused me, when I was testing M2_HOME didn't seem to be set. If it works great!

**Review by k-wall — APPROVED** [review]
Please remember to squash and give it a sensible commit message

**Pull request merged**
